crates/primitives-traits/src/extended.rs

```diff
@@ -303,59 +297,73 @@ where
     }
 }
 
-#[derive(Debug, serde::Serialize, serde::Deserialize)]
-pub enum ExtendedTxEnvelopeRepr<'a, B: SerdeBincodeCompat, T: SerdeBincodeCompat> {
-    BuiltIn(B::BincodeRepr<'a>),
-    Other(T::BincodeRepr<'a>),
-}
+#[cfg(feature = "serde-bincode-compat")]
+mod serde_bincode_compat {
+    use super::*;
+    use crate::serde_bincode_compat::SerdeBincodeCompat;
 
-impl<B, T> SerdeBincodeCompat for ExtendedTxEnvelope<B, T>
-where
-    B: SerdeBincodeCompat + std::fmt::Debug,
-    T: SerdeBincodeCompat + std::fmt::Debug,
-{
-    type BincodeRepr<'a> = ExtendedTxEnvelopeRepr<'a, B, T>;
+    #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+    #[derive(Debug)]
+    pub enum ExtendedTxEnvelopeRepr<'a, B: SerdeBincodeCompat, T: SerdeBincodeCompat> {
+        BuiltIn(B::BincodeRepr<'a>),
+        Other(T::BincodeRepr<'a>),
+    }
 
-    fn as_repr(&self) -> Self::BincodeRepr<'_> {
-        match self {
-            Self::BuiltIn(tx) => ExtendedTxEnvelopeRepr::BuiltIn(tx.as_repr()),
-            Self::Other(tx) => ExtendedTxEnvelopeRepr::Other(tx.as_repr()),
+    impl<B, T> SerdeBincodeCompat for ExtendedTxEnvelope<B, T>
+    where
+        B: SerdeBincodeCompat + std::fmt::Debug,
+        T: SerdeBincodeCompat + std::fmt::Debug,
+    {
+        type BincodeRepr<'a> = ExtendedTxEnvelopeRepr<'a, B, T>;
+
+        fn as_repr(&self) -> Self::BincodeRepr<'_> {
+            match self {
+                Self::BuiltIn(tx) => ExtendedTxEnvelopeRepr::BuiltIn(tx.as_repr()),
+                Self::Other(tx) => ExtendedTxEnvelopeRepr::Other(tx.as_repr()),
+            }
         }
-    }
 
-    fn from_repr(repr: Self::BincodeRepr<'_>) -> Self {
-        match repr {
-            ExtendedTxEnvelopeRepr::BuiltIn(tx_repr) => Self::BuiltIn(B::from_repr(tx_repr)),
-            ExtendedTxEnvelopeRepr::Other(tx_repr) => Self::Other(T::from_repr(tx_repr)),
+        fn from_repr(repr: Self::BincodeRepr<'_>) -> Self {
+            match repr {
+                ExtendedTxEnvelopeRepr::BuiltIn(tx_repr) => Self::BuiltIn(B::from_repr(tx_repr)),
+                ExtendedTxEnvelopeRepr::Other(tx_repr) => Self::Other(T::from_repr(tx_repr)),
+            }
         }
     }
 }
 
-impl<B, T> Compact for ExtendedTxEnvelope<B, T>
-where
-    B: Transaction + IsTyped2718 + Compact,
-    T: Transaction + Compact,
-{
-    fn to_compact<Buf>(&self, buf: &mut Buf) -> usize
+#[cfg(feature = "reth-codec")]
+mod compact {
```

Comment:
I'd like to avoid this compact module and use full paths instead, this is how we do compact feature gate elsewhere